Zend/zend_ast.c

```diff
@@ -1372,13 +1373,24 @@ static ZEND_COLD void zend_ast_export_attributes(smart_str *str, zend_ast *ast,
 			}
 			smart_str_appendc(str, ')');
 		}
+	}
+}
 
-		if (newlines) {
-			smart_str_appendc(str, '\n');
-			zend_ast_export_indent(str, indent);
-		} else {
-			smart_str_appendc(str, ' ');
-		}
+static ZEND_COLD void zend_ast_export_attributes(smart_str *str, zend_ast *ast, int indent, zend_bool newlines) {
+	zend_ast_list *list = zend_ast_get_list(ast);
+	uint32_t i;
+
+	for (i = 0; i < list->children; i++) {
+		smart_str_appends(str, "#[");
+		zend_ast_export_attribute_group(str, list->child[i], indent);
+		smart_str_appends(str, "]");
+	}
+
+	if (newlines) {
```

Comment:
The newline / blank insertion needs to be appended to the `for` loop instead. It will insert a newline / blank after each attribute group to generate consistent output.